syntax = "proto3";

message Data {
  map<string, string> entry = 1;
}

message PutRequest {
    string id = 1;
    Data data = 2;
}

message PutResponse {
    string id = 1;
}

message GetRequest {
    string id = 1;
}

message GetResponse {
    Data data = 1;
}

message Empty {}

message InfoResponse {
    Data data = 1;
}

service DB {
    rpc put(PutRequest) returns (PutResponse) {}
    rpc get(GetRequest) returns (GetResponse) {}
    rpc info(Empty) returns (InfoResponse) {}
}